kvmagent/kvmagent/plugins/vms/vm_host_file.py

```
@@ -0,0 +1,139 @@
+    os.makedirs(extract_dir)
+
+    try:
+        with open(tmp_tar_file, 'wb') as f:
+            f.write(raw_data)
+            f.flush()
+            os.fsync(f.fileno())
+
+        cmd = "tar -xzf %s -C %s" % (tmp_tar_file, extract_dir)
+        r, _, e = bash.bash_roe(cmd) 
+        if r != 0:
+            raise RuntimeError("Failed to untar data: %s, stderr: %s" % (cmd, e))
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)